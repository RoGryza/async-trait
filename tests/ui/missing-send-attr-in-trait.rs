use async_trait::async_trait;

#[async_trait]
pub trait Trait {
    async fn method();
}

pub struct Struct;

#[async_trait]
impl Trait for Struct {
    #[async_trait(?Send)]
    async fn method() {}
}

fn main() {}
