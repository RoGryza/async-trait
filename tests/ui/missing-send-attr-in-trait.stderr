error[E0053]: method `method` has an incompatible type for trait
  --> tests/ui/missing-send-attr-in-trait.rs:13:5
   |
13 |     async fn method() {}
   |     ^^^^^^^^^^^^^^^^^ expected trait `Future<Output = ()> + Send`, found trait `Future<Output = ()>`
   |
note: type in trait
  --> tests/ui/missing-send-attr-in-trait.rs:5:5
   |
5  |     async fn method();
   |     ^^^^^^^^^^^^^^^^^^
   = note: expected fn pointer `fn() -> Pin<Box<(dyn Future<Output = ()> + Send + 'async_trait)>>`
              found fn pointer `fn() -> Pin<Box<(dyn Future<Output = ()> + 'async_trait)>>`
